pkg/operator/ceph/cluster/cleanup.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mgr"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/rbd"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/ceph/file/mds"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/util"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	clusterCleanUpPolicyRetryInterval = 5 //seconds
+	// CleanupAppName is the cluster clean up job name
+	CleanupAppName = "rook-ceph-cleanup"
+)
+
+var (
+	volumeName      = "cleanup-volume"
+	dataDirHostPath = "ROOK_DATA_DIR_HOST_PATH"
+	namespaceDir    = "ROOK_NAMESPACE_DIR"
+)
+
+func (c *ClusterController) startClusterCleanUp(cluster *cephv1.CephCluster, cephHosts []string) {
+	logger.Infof("starting clean up for cluster %q", cluster.Name)
+	err := c.waitForCephDaemonCleanUp(cluster, time.Duration(clusterCleanUpPolicyRetryInterval)*time.Second)
+	if err != nil {
+		logger.Errorf("failed to wait till ceph daemons are destroyed. %v", err)
```

Comment:
Why not using `wrap` error syntax here?

Reply:
`wrap` should be used when returning the error, right? Here we're just logging the error.

Reply:
Oh. :woman_facepalming: